main/accounts/types.ts

```diff
@@ -31,21 +31,17 @@ export enum RequestStatus {
   Success = 'success'
 }
 
-type RequestType =
-  | 'sign'
-  | 'signTypedData'
-  | 'transaction'
-  | 'access'
-  | 'addChain'
-  | 'switchChain'
-  | 'addToken'
+export type SignatureRequestType = 'sign' | 'signTypedData' | 'signErc20Permit'
+
+type RequestType = SignatureRequestType | 'transaction' | 'access' | 'addChain' | 'switchChain' | 'addToken'
 
 export interface AccountRequest {
   type: RequestType
   origin: string
   payload: JSONRPCRequestPayload
   handlerId: string
   account: string
+  typedMessage?: TypedMessage<SignTypedDataVersion>
```

Comment:
do we need this on the base request? I dont think all requests have this attribute